probeinterface/io.py

```diff
@@ -99,6 +99,292 @@ def write_probeinterface(file, probe_or_probegroup):
+                          in v.keys()]
+        contact_params = np.unique(contact_params)
+
+        # collect contact information for each probe_id
+        for probe in probes.values():
+            contact_ids = probe.electrode_ids
+            for contact_param in contact_params:
+                # collect parameters across contact id to add to probe
+                value_list = [
+                    contacts_dict['contact_id'][str(c)].get(contact_param, None)
+                    for c in contact_ids]
+
+                if contact_param == 'contact_shape_params':
+                    probe.electrode_shape_params = value_list
+                elif contact_param == 'contact_plane_axes':
+                    probe.electrode_plane_axes = np.array(value_list)
+
+                # extract units if this was not extracted earlier already
+                elif contact_param == 'contact_shape_units' and \
+                        all(x == value_list[0] for x in value_list):
+                    if probe.si_units == None:
+                        probe.si_units = value_list[0]
+                    elif probe.si_units != value_list[0]:
+                        raise ValueError(f'Different units used for probe and '
+                                         f'contacts ({probe.si_units} != '
+                                         f'{value_list[0]})')
+                else:
+                    probe.annotate(**{contact_param: value_list})
+
+    for probe in probes.values():
+        probegroup.add_probe(probe)
+
+    return probegroup
+
+
+def write_BIDS_probe(file, probe_or_probegroup):
+    """
+    Write to probe and contact formats as proposed
+    for ephy BIDS extension (tsv & json based).
+
+    The format handles several probes in one file.
+
+    Parameters
+    ----------
+    file: Path or str
+        The folder name.
+
+    probe_or_probegroup : Probe or ProbeGroup
+        If probe is given a probegroup is created anyway.
+
+    """
+    import pandas as pd
+
+    if isinstance(probe_or_probegroup, Probe):
+        probe = probe_or_probegroup
+        probegroup = ProbeGroup()
+        probegroup.add_probe(probe)
+    elif isinstance(probe_or_probegroup, ProbeGroup):
+        probegroup = probe_or_probegroup
+    else:
+        raise ValueError('probe_or_probegroup has to be'
+                         'of type Probe or ProbeGroup')
+    file = Path(file)
+    probes = probegroup.probes
+
+    def get_unique_ids(min, max, n, trials=20):
+        """Create n unique ids between min and max within a maximum number of
+        trials (attempts)"""
+        ids = np.random.randint(min, max, n)
+        i = 0
+
+        while len(np.unique(ids)) != len(ids) and i < trials:
+            ids = np.random.randint(min, max, n)
+
+        if len(np.unique(ids)) != len(ids):
+            raise ValueError(f'Can not generate {n} unique ids between {min} '
+                             f'and {max} in {trials} trials')
+        return ids
+
```

Comment:
Instead of generating this in this function we could add a auto_generate_ids in ProbeGroup. 
Then here we could raise an error if the probegroup don't have electrode_ids with a message
"this probe group don't have ids, you should generate with ..."

Reply:
Yes, this would be useful to add to the main functionality of probeinterface! For now I just generate unique integers, do yo have another preferred format for probe ids?

Reply:
This functionality is now implemented on the probegroup level.